tensorflow/contrib/grid_rnn/python/kernel_tests/grid_rnn_test.py

```diff
@@ -0,0 +1,324 @@
+# Copyright 2015 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+"""Tests for GridRNN cells."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+import tensorflow as tf
+
+
+class GridRNNCellTest(tf.test.TestCase):
+
+  def testGrid2BasicLSTMCell(self):
+    with self.test_session() as sess:
+      with tf.variable_scope('root', initializer=tf.constant_initializer(0.2)) as root_scope:
+        x = tf.zeros([1, 3])
+        m = tf.zeros([1, 8])
+        cell = tf.contrib.grid_rnn.Grid2BasicLSTMCell(2, input_size=3)
+        self.assertEqual(cell.output_size, 2)
+        self.assertEqual(cell.input_size, 3)
+        self.assertEqual(cell.state_size, 8)
+        g, s = cell(x, m)
+        sess.run([tf.initialize_all_variables()])
+        res = sess.run([g, s], {x.name: np.array([[1., 1., 1.]]),
+                             m.name: np.array([[0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8]])})
+        self.assertEqual(res[0].shape, (1, 2))
+        self.assertEqual(res[1].shape, (1, 8))
+        self.assertAllClose(res[0], [[ 0.36617181, 0.36617181]])
```

Comment:
how are you getting these numbers?  are they just a smoke test?

if so, convert all these tests to just making sure that they're not nan/inf and verifying that the shapes are correct.

also add checks on g.get_shape() and s.get_shape()